packages/@react-aria/dnd/src/DragManager.ts

```diff
@@ -197,7 +206,19 @@ class DragSession {
     );
     this.updateValidDropTargets();
 
-    announce(this.stringFormatter.format(MESSAGES[getDragModality()]));
+    let dragModality = getDragModality();
+    announce(this.stringFormatter.format(MESSAGES[dragModality]));
+
+    if (dragModality === 'virtual') {
```

Comment:
Starting In forms mode, it will be “keyboard”, but starting in browse/virtual cursor mode, it will be “virtual,” and this is when we need to force forms mode.

Reply:
I've done some further testing and it seems that something has changed with keyboard interaction for drag and drop since this PR was opened, and it is no longer possible to initiate a Drag and Drop operation using `Enter` key with NVDA or JAWS unless you are already in forms mode or you also hold you hold down the `Shift` key when you press `Enter`. You can reproduce this failure using the latest build on main branch: https://reactspectrum.blob.core.windows.net/reactspectrum/9aeb1726dcd040565035fa94c41083a966d337a3/storybook/index.html?path=/story/drag-and-drop--default&providerSwitcher-toastPosition=bottom. 

I'm not entirely sure, but I wonder if this regression was introduced by https://github.com/adobe/react-spectrum/pull/3421.

Reply:
I can verify that things work as expected, dragging starts and forms mode is forced on when Enter is pressed on the drag target button, using this earlier version from Aug 16, before #3421 was merged into main and then this PR: https://reactspectrum.blob.core.windows.net/reactspectrum/92861d8a2f148ef08801d02365b88af90247ced0/storybook/index.html?path=/story/drag-and-drop--default&providerSwitcher-toastPosition=bottom. After #3421, the default Drag and Drop story does not work if one tries to press Enter on the drag target button in browse/virtual cursor mode.